Failing tests on CI

Currently, the Android Geocoder test always fails on CI, e.g. http://ci.cgeo.org/job/cgeo%20pull%20request/1444/testReport/cgeo.geocaching.address/GeocoderTest/testAndroidGeocoder/

In a [comment](https://github.com/cgeo/cgeo/pull/7149#issuecomment-425712144) @rsudev stated that this is due to a request limit but I don't think that's the case. I think we receive the requested address data but it just does not contain the country code anymore.

So we should check whether the country code is missing in general and either remove the country code check or use other test data that contains a country code.

`website tests` fixed, but continuous integration failed with a new error:

```
07:31.385  cgeo.geocaching.connector.su.SUParserTest > testCanHandle FAILED
07:31.385      java.lang.ExceptionInInitializerError at SUParserTest.java:34
07:31.385          Caused by: java.lang.NullPointerException at SUParserTest.java:34
```

The stack trace in the [log](http://ci.cgeo.org/job/cgeo%20continuous%20integration/2040/testReport/cgeo.geocaching.connector.su/SUParserTest/testCanHandle/) is incomplete:
```
Caused by: java.lang.NullPointerException
	at cgeo.geocaching.settings.Settings.<clinit>(Settings.java:104)
	... 54 more
```

--- a/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
@@ -87,8 +87,7 @@ public final class MapsforgeMapProvider extends AbstractMapProvider {
 
         try {
             final MapFile mapFile = new MapFile(mapFileIn);
-            if (mapFile.getMapFileInfo().fileVersion > 3 && Settings.useOldMapsforgeAPI()) return false;
-            return true;
+            return mapFile.getMapFileInfo().fileVersion <= 3 || !Settings.useOldMapsforgeAPI();
         } catch (MapFileException ex){
             Log.w(String.format("Exception reading mapfile '%s'", mapFileIn), ex);
         }

--- a/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
@@ -30,8 +30,6 @@ import org.mapsforge.map.model.MapViewPosition;
 import org.mapsforge.map.reader.MapFile;
 import org.mapsforge.map.reader.header.MapFileException;
 import org.mapsforge.v3.android.maps.mapgenerator.MapGeneratorInternal;
-import org.mapsforge.v3.map.reader.MapDatabase;
-import org.mapsforge.v3.map.reader.header.FileOpenResult;
 
 public final class MapsforgeMapProvider extends AbstractMapProvider {
 

--- a/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapProvider.java
@@ -88,7 +88,7 @@ public final class MapsforgeMapProvider extends AbstractMapProvider {
         try {
             final MapFile mapFile = new MapFile(mapFileIn);
             return mapFile.getMapFileInfo().fileVersion <= 3 || !Settings.useOldMapsforgeAPI();
-        } catch (MapFileException ex){
+        } catch (MapFileException ex) {
             Log.w(String.format("Exception reading mapfile '%s'", mapFileIn), ex);
         }
         return false;

--- a/tests/src-android/cgeo/geocaching/connector/su/SUParserTest.java
+++ b/tests/src-android/cgeo/geocaching/connector/su/SUParserTest.java
@@ -8,8 +8,9 @@ import org.junit.Test;
 import java.io.InputStream;
 
 import cgeo.geocaching.SearchResult;
+import cgeo.geocaching.test.AbstractResourceInstrumentationTestCase;
 
-public class SUParserTest {
+public class SUParserTest extends AbstractResourceInstrumentationTestCase {
 
     @Test
     public void testCanHandle() throws Exception {

